daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala

```diff
@@ -18,6 +18,9 @@ import scala.language.implicitConversions
 @SuppressWarnings(Array("org.wartremover.warts.Any"))
 class HashSpec extends WordSpec with Matchers {
 
+  @com.github.ghik.silencer.silent // dead code. Well, yeah
```

Comment:
not clear from the context where it is being used and why it is needed.

Reply:
> not clear from the context where it is being used

The scope of the annotation is the following definition, which is one line long.

> and why it is needed.

The warning is a dead code warning. How many lines of comments should I devote to detailing it?

Reply:
I did not mean the annotation... I meant it is not clear why we need `scalaz.Order[Nothing]`

Reply:
Well, the short version is, "the code will not compile without it".

The long version: this sort of thing happens in systems with subtyping. Let's say that you want to define list comparison inductively, i.e. `[A] Order[A] => Order[List[A]]`. Then, let's say that you want a property of your ordering to be that "if all of `xs...` and `ys...` expressions unify to a comparable type, then `List(xs...) ?|? List(ys...)` (where `?|?` is comparison) is well-typed."

If `xs...` and `ys...` are empty, then they unify to `Nothing`. And it is perfectly sensible to expect `List() ?|? List()` to be well-typed, even though the only reasonable type of the arguments is `List[Nothing]`. (`List[Any]` is *not* a reasonable choice, because it literally fits nothing but itself, whereas `List[Nothing]` fits anything.) But the only way to get `List[Nothing]` to have an `Order` is, by reading our induction backwards, to have an `Order[Nothing]`.

In our data model, the spiritual equivalent of "empty list", "value with no contract IDs", is quite common, so it is quite likely for the unified contract ID type to be `Nothing`.

The good news is, unlike the other `Order` instance I define in this file, the `Order[Nothing]` is guaranteed to be correct as declared; the way we interpret the principle of explosion here is "if an argument's type is `Nothing` then literally all implementations are correct".

I seriously considered several times defining a context unifier for `IntroCtx` to prove implicitly those cases that would not need an `Order`, but decided that this was not worth the maintenance (or code review) burden; thus, it is simply defined as an alias for `Order`, causing this instance to be needed.